components/gitpod-protocol/src/analytics.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+export const IAnalyticsWriter = Symbol("IAnalyticsWriter");
+
+type Identity =
+    | { userId: string | number }
+    | { userId?: string | number; anonymousId: string | number };
+
+interface Message {
+    messageId?: string;
```

Comment:
What is the messageId and its purpose? Can it be omitted on the client-side and generically added when forwarded to segment?

Reply:
`messageId` is not required per se, but can be used to de-duplicate messages. Segment will de-dup messages with the same ID that came in in a 24h window.